sunpy/data/__init__.py

```diff
@@ -1,3 +1,41 @@
+import astropy.units as u
+import sys
+
+from sunpy import config
 from sunpy.data._sample import download_sample_data
+from sunpy.data.data_manager.cache import Cache
+from sunpy.data.data_manager.downloader import ParfiveDownloader
+from sunpy.data.data_manager.manager import DataManager
+from sunpy.data.data_manager.storage import SqliteStorage
+from sunpy.util.config import CACHE_DIR
+
+_download_dir = config.get('downloads', 'remote_data_manager_dir')
+
+
+if 'pytest' in sys.modules:
+    from sunpy.data.data_manager.storage import InMemStorage
+    import tempfile
+    cache_dir = tempfile.mkdtemp()
+    cache = Cache(
+        ParfiveDownloader(),
+        InMemStorage(),
+        cache_dir,
+        None
+    )
+    manager = DataManager(cache)
```

Comment:
This entire block is setup for the test suite?

Reply:
Yes. I tried patching the object using conftest but that wasn't working.

Reply:
Then do you still need the functions you added in conftest?

Reply:
Yes, those are for adding local file to data manager. That can be useful for mocking remote data tests and ideally all tests should use that.

Reply:
Ah ok, could you write a comment saying that somewhere?

Reply:
You haven't added a comment about this yet.

Also the `sys.modules` check makes me rather nervous here, doesn't that mean that if someone imports pytest before `sunpy.data` they will hit this code path?

What are you working around here maybe I can think of another way out?

Reply:
> Also the sys.modules check makes me rather nervous here, doesn't that mean that if someone imports pytest before sunpy.data they will hit this code path?

This might be true, but I couldn't find another way.

> What are you working around here maybe I can think of another way out?

When we are running tests, we don't want the cache to interfere with the tests. So this installs a mock cache in the place of the real cache. 

I tried patching the cache in the conftest but that won't work due to a variety of reasons.

I specifically made this change because the tests were erroring due to some reason I can't remember now. It was more or less caused by cache interfering with the tests.